pkg/k8shandler/cluster.go

```diff
@@ -207,16 +209,64 @@ func getNodeUpgradeInProgress(cluster *v1alpha1.Elasticsearch) NodeTypeInterface
 	return nil
 }
 
+func setUUIDs(cluster *v1alpha1.Elasticsearch) {
+
+	for index := 0; index < len(cluster.Spec.Nodes); index++ {
+		if cluster.Spec.Nodes[index].GenUUID == nil {
+			uuid, err := utils.RandStringBytes(8)
+			if err != nil {
+				continue
+			}
+
+			// update the node to set uuid
+			cluster.Spec.Nodes[index].GenUUID = &uuid
+
+			nretries := -1
+			retryErr := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+				nretries++
+				if getErr := sdk.Get(cluster); getErr != nil {
+					logrus.Debugf("Could not get Elasticsearch %v: %v", cluster.Name, getErr)
+					return getErr
+				}
+
+				if cluster.Spec.Nodes[index].GenUUID != nil {
+					return nil
+				}
+
+				cluster.Spec.Nodes[index].GenUUID = &uuid
+
+				if updateErr := sdk.Update(cluster); updateErr != nil {
+					logrus.Debugf("Failed to update Elasticsearch %s status. Reason: %v. Trying again...", cluster.Name, updateErr)
+					return updateErr
+				}
+				return nil
+			})
+
+			if retryErr != nil {
+				logrus.Errorf("Error: could not update status for Elasticsearch %v after %v retries: %v", cluster.Name, nretries, retryErr)
+			}
+			logrus.Debugf("Updated Elasticsearch %v after %v retries", cluster.Name, nretries)
+		}
+	}
+
+}
+
 func getNodes(cluster *v1alpha1.Elasticsearch) {
 
+	setUUIDs(cluster)
```

Comment:
This seems like an odd place to to set the UUID given we are trying to retrieve the nodes.  Shouldn't we do this when creating the deployment objects the first time

Reply:
we need to make sure that each node has a UUID before we go to create it, it simplifies the logic so that we can just generate the name based on the UUID.

Also `getNodes` might be poorly named, its populating the `nodes` map based on the `Spec.Nodes` objects